src/utils/perfagg.py

```diff
@@ -86,6 +87,133 @@
 }
 
 
+# joins disparate runs less dumbly than rocprof
+def join_prof(workload_dir, join_type, out=None):
+    # Set default output directory if not specified
+    if out == None:
+        out = workload_dir + "/pmc_perf.csv"
+    files = glob.glob(workload_dir + "/" + "pmc_perf_*.csv")
+    df = None
+
+    for i, file in enumerate(files):
+        # _df = parse_rocprof_kernels(file)
+
+        _df = pd.read_csv(file)
+        key = _df.groupby("KernelName").cumcount()
+        if join_type == "kernel":
+            _df["key"] = _df.KernelName + " - " + key.astype(str)
+        elif join_type == "grid":
+            _df["key"] = (
+                _df.KernelName + " - " + key.astype(str) + " - " + _df.grd.astype(str)
+            )
+        else:
+            print("ERROR: Unrecognized --join-type")
+            sys.exit(1)
+
+        if df is None:
+            df = _df
+        else:
+            # join by unique index of kernel
+            df = pd.merge(df, _df, how="inner", on="key", suffixes=("", f"_{i}"))
+    # now, we can:
+    #   A) throw away any of the "boring" duplicats
+    df = df[
+        [
+            k
+            for k in df.keys()
+            if not any(
+                check in k
+                for check in [
+                    # "gpu",
+                    "queue-id",
+                    "queue-index",
+                    "pid",
+                    "tid",
+                    # "grd",
+                    # "wgr",
+                    # "lds",
+                    # "scr",
+                    # "vgpr",
+                    # "sgpr",
+                    "fbar",
+                    "sig",
+                    "obj",
+                ]
+            )
+        ]
+    ]
+    #   B) any timestamps that are _not_ the duration, which is the one we care
+    #   about
+    df = df[
+        [
+            k
+            for k in df.keys()
+            if not any(check in k for check in ["DispatchNs", "CompleteNs"])
+        ]
+    ]
+    #   C) sanity check the name and key
+    namekeys = [k for k in df.keys() if "KernelName" in k]
+    assert len(namekeys)
+    for k in namekeys[1:]:
+        assert (df[namekeys[0]] == df[k]).all()
+    df = df.drop(columns=namekeys[1:])
+    # now take the median of the durations
+    bkeys = []
+    ekeys = []
+    for k in df.keys():
+        if "Begin" in k:
+            bkeys.append(k)
+        if "End" in k:
+            ekeys.append(k)
+    # compute mean begin and end timestamps
+    endNs = df[ekeys].mean(axis=1)
+    beginNs = df[bkeys].mean(axis=1)
+    # and replace
+    df = df.drop(columns=bkeys)
+    df = df.drop(columns=ekeys)
+    df["BeginNs"] = beginNs
+    df["EndNs"] = endNs
+    # finally, join the drop key
+    df = df.drop(columns=["key"])
+    # and save to file
+    df.to_csv(out, index=False)
+    # and delete old file(s)
+    for file in files:
+        os.remove(file)
```

Comment:
It occurs to me, we may want to leave the old files in place in VERBOSE mode (I think is the name? can't remember -- the one you added the other week that keeps more logging information around)

Reply:
https://github.com/AMDResearch/omniperf/blob/44edef4635c18ee3f2bb764f7078db05dfbbecd6/src/utils/perfagg.py#L211-L214